### DISCUSSION TITLE
Release 5.1.0 Vidyadaan Deployment issues

### DISCUSSION THREAD

**Discussion by sandeep-r1** [Sunbird-Ed/Community#365] [Bugs] [answered]
Hi Team,
Below are the jobs are failed while deploying

**1)PostgresqlMigration**
`failed: [localhost] (item=) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "docker run --env-file=/var/lib/jenkins/workspace/Deploy/DockPreproduction/Kubernetes/PostgresqlMigration/ansible/roles/postgres-migration/flyway.env --network=host --rm -v /var/lib/jenkins/workspace/Deploy/DockPreproduction/Kubernetes/PostgresqlMigration/ansible/roles/postgres-migration/files/:/flyway/sql flyway/flyway:7.13.0 -url=jdbc:postgresql://dock-preprod-all.postgres.database.azure.com/ -user=dockpreprod@dock-preprod-all migrate",
    "delta": "0:00:07.967482",
    "end": "2023-01-20 23:32:37.183145",
    "item": "",
    "rc": 1,
    "start": "2023-01-20 23:32:29.215663"
}

STDERR:

ERROR: Unable to obtain connection from database (jdbc:postgresql://dock-preprod-all.postgres.database.azure.com/) for user 'dockpreprod@dock-preprod-all': FATAL: Invalid connection parameters specified. Check connection parameters, and try again.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 28000
Error Code : 0
Message    : FATAL: Invalid connection parameters specified. Check connection parameters, and try again.

Caused by: org.postgresql.util.PSQLException: FATAL: Invalid connection parameters specified. Check connection parameters, and try again.

MSG:

non-zero return code`

**2)Content**
`changed: [localhost] => (item=main.rego)
failed: [localhost] (item=common.rego) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "common.rego"
}

MSG:

AnsibleUndefinedVariable: 'adminutil_refresh_token_public_key_kid' is undefined`

**3)Taxonomy**
`It is taking more time while doing  helm install task`

**4)OnboardAPI**
`fatal: [localhost]: FAILED! => {
    "changed": true,
    "cmd": "python /tmp/kong-api-scripts/kong_apis.py /tmp/kong_apis.json --kong-admin-api-url=[http://11.6.16.68/admin-api",](http://11.6.16.68/admin-api%22,%1B[0m)
    "delta": "0:00:30.224225",
    "end": "2023-01-20 22:57:23.231385",
    "rc": 1,
    "start": "2023-01-20 22:56:53.007160"
}

STDOUT:

Number of input APIs : 471
Number of existing APIs : 389
Input APIs : [u'acceptContentFlag', u'acceptTermsAndCondition', u'addBadgeToUser', u'addSystemSettings', u'assemblePage', u'assignRole', u'assignRoleV2', u'batchBulkEnrolment', u'batchBulkUnEnrolment', u'blockUser', u'checkUploadJobStatus', u'compositeSearch', u'compositePrivateSearch', u'contentBadgeLink', u'contentBadgeUnLink', u'copyContent', u'copyFramework', u'courseBatchAddCertificateTemplate', u'courseBatchRemoveCertificateTemplate', u'courseEnrollment', u'courseIssueCertificate', u'courseUnEnrollment', u'createBatch', u'createCertTemplate', u'createChannel', u'createContent', u'createCourse', u'createForm', u'createFramework', u'createFrameworkCategory', u'createFrameworkTerm', u'createLock', u'createNote', u'createOrg', u'createPage', u'createPageSection', u'createPublisher', u'createSession', u'createUser', u'createUserLearnerV3', u'createUserVersion2', u'createUserVersion3', u'createUserVersion4', u'deleteNote', u'deviceProfile', u'deviceRegister', u'dialAssemble', u'downloadCertificate', u'downloadRegCertificate', u'downloadTOC', u'echo', u'experimentcreate', u'experimentread', u'fetchURLMetaInformation', u'flagContent', u'freshDesk', u'generateDialcodes', u'generateOtp', u'generateOtpV2', u'getBatch', u'collectionSummaryAgg', u'getContentUploadUrl', u'getCourseHierarchy', u'getPageSettings', u'getParticipants', u'getSystemSettings', u'getTenantInfo', u'getUserByKey', u'getUserByKeyV2', u'getUserProfile', u'getUserProfileV2', u'indexSync', u'ItemSetAPIs', u'KongConsumerApi', u'LicenseAPI', u'linkDialcodeContent', u'listAllBadges', u'listChannel', u'searchCourseBatches', u'listCourseEnrollments', u'privateListCourseEnrollments', u'listUserCourseEnrollments', u'privateListUserCourseEnrollments', u'listDialcodes', u'listFramework', u'listLock', u'listOrdinals', u'listPageSections', u'listResourceBundles', u'listSystemSettings', u'listTerms', u'masterLocationCreate', u'masterLocationDelete', u'masterLocationSearch', u'masterLocationUpdate', u'masterLocationUpload', u'mergeUserAccounts', u'orgAssignKeys', u'orgBulkUpload', u'pluginSearch', u'PrivateContentCreateAPIs', u'PrivateContentReadAPIs', u'PrivateContentUpdateAPIs', u'PrivateContentRetireAPI', u'PrivateContentUploadAPI', u'PrivateContentReviewAPI', u'PrivateContentPublishAPI', u'PrivateContentRejectAPI', u'PrivateContentHierarchyAPI', u'privateUserAssignRole', u'privateUserRead', u'publishContent', u'publishCourse', u'publishDialcode', u'publishFramework', u'qrCodeBatchProcessRetry', u'qrCodeBatchProcessStatus', u'qrCodeDownload', u'readCertTemplate', u'readChannel', u'readContent', u'contentPrivateRead', u'readContentState', u'privateReadContentState', u'readCourse', u'readCourseEnrollment', u'readDialcode', u'readDialcodeV2', u'readForm', u'readFramework', u'readFrameworkCategory', u'readFrameworkTerm', u'readNote', u'readOrg', u'readPageSection', u'readPublisher', u'readRoleMapping', u'readUserSpecificRole', u'readUserProfile', u'registerDesktopApp', u'registerDesktopDevice', u'registerDesktopDevicev2', u'refreshLock', u'refreshToken', u'registerMobileApp', u'registerMobileAppOpenRAP', u'registerMobileAppTeacherAid', u'registerMobileDevice', u'registerMobileDevicev2', u'registerPortalAnonymous', u'registerPortalLoggedin', u'registerMobileDeviceOpenRAP', u'registerMobileDeviceTeacherAid', u'rejectContent', u'rejectContentFlag', u'releaseDialcode', u'reserveDialcode', u'retireContent', u'retireCourse', u'retireLock', u'searchChannel', u'searchContent', u'searchCourse', u'searchDialcodes', u'searchFrameworkCategory', u'searchFrameworkTerm', u'searchNotes', u'searchOrgExtended', u'searchOrg', u'searchUser', u'sendEmailNotification', u'submitContentForReview', u'submitCourseForReview', u'telemetry', u'telemetryErrorLogging', u'unblockUser', u'updateBatch', u'updateCertTemplate', u'updateChannel', u'updateCollaborators', u'updateContent', u'updateContentState', u'updateCourse', u'linkDialcodeContentV1', u'reserveDialcodeContent', u'updateCourseHierarchy', u'updateDesktopApp', u'updateDialcode', u'updateDialcodeV2', u'updateForm', u'updateFramework', u'updateFrameworkCategory', u'updateFrameworkTerm', u'updateNote', u'updateOrg', u'updateOrgStatus', u'updatePage', u'updatePageSection', u'updatePublisher', u'updateUser', u'uploadContent', u'uploadCrashLogs', u'uploadTOC', u'userBulkUpload', u'userExistenceApi', u'userExistenceApiV2', u'userFeed', u'userFeedCreate', u'userFeedDelete', u'userFeedUpdate', u'userMigrate', u'validateRegCertificate', u'verifyOtp', u'verifyOtpV2', u'verifyRegCertificate', u'registerIgotApp', u'registerIgotDevice', u'listReports', u'getReport', u'updateReport', u'deleteReport', u'createReport', u'createReportSummary', u'getReportSummary', u'listReportSummary', u'searchRegCertificate', u'searchRegCertificateV2', u'readRegCertificate', u'forgotPassword', u'publishReport', u'retireReport', u'searchManagedUser', u'createGroup', u'updateGroup', u'listGroup', u'readGroup', u'deleteGroup', u'updateGroupMembership', u'groupActivityAgg', u'listForm', u'getUserProfileV3', u'updateUserDeclarations', u'updateUserConsent', u'readUserConsent', u'readUserConsentV2', u'channelSpecificTelemetryExhaust', u'publicDataExhaust', u'submitDataExhaustRequest', u'getDataExhaustRequest', u'listDataExhaustRequest', u'addDataset', u'listDataset', u'createTenantPreferences', u'updateTenantPreferences', u'readTenantPreferencess', u'generateCertificateV2', u'addRegCertificateV2', u'downloadRegCertificateV2', u'contentImportAPI', u'druidGetUpdateListReport', u'druidDeactivateReport', u'druidCreateReport', u'addRegCertificate', u'generateCertificate', u'createObjectCategory', u'readObjectCategory', u'updateObjectCategory', u'createObjectCategoryDefinition', u'readObjectCategoryDefinition', u'updateObjectCategoryDefinition', u'createAsset', u'updateAsset', u'readAsset', u'uploadAsset', u'uploadUrlAsset', u'copyAsset', u'collectionCreate', u'collectionRead', u'collectionPrivateRead', u'assessmentItemRead', u'collectionUpdate', u'collectionFlag', u'collectionDiscard', u'collectionRetire', u'linkDialcodeCollection', u'collectionHierarchyAdd', u'collectionHierarchyRemove', u'collectionHierarchyUpdate', u'collectionHierarchyRead', u'collectionCopy', u'collectionImport', u'collectionExport', u'getDiscussionTagsList', u'getDiscussionCategories', u'getDiscussionNotificationsList', u'getUserDetailsOfDiscussion', u'getCategoryDetailsOfDiscussion', u'getUnreadTopicsOfDiscussion', u'getRecentTopicsOfDiscussion', u'getPopularTopicsOfDiscussion', u'getTopTopicsOfDiscussion', u'getTopicsOfDiscussionById', u'getTotalUnreadTopicsOfDiscussion', u'getTopicsOfDiscussionByTeaserId', u'getTopicsPaginationByIdOfDiscussion', u'getGroupsListOfDiscussion', u'getRecentPostsByDateOfDiscussions', u'createTopicOfDiscussions', u'createCategoryOfDiscussion', u'createGroupsOfDiscussion', u'createNewPostOfDiscussion', u'createNewUserOfDiscussion', u'getUserDetailsByUsername', u'getForumIdOfDiscussion', u'addForumOfDiscussion', u'copyPrivilegesFromParentCategory', u'questionCreate', u'questionRead', u'questionPrivateRead', u'questionUpdate', u'questionRetire', u'questionReview', u'questionPublish', u'questionList', u'questionReject', u'questionSetCreate', u'questionSetRead', u'questionSetPrivateRead', u'questionSetUpdate', u'questionSetAddQuestion', u'questionSetRemoveQuestion', u'questionSetRetire', u'questionSetReview', u'questionSetPublish', u'questionSetUpdateHierarchy', u'questionSetReadHierarchy', u'questionSetReject', u'updateUserV2', u'questionImportAPI', u'managedUserV1Create', u'managedUserV2Create', u'ssoUserV1Create', u'ssoUserV2Create', u'getUserProfileV4', u'getUserProfileV5', u'searchUserV2', u'searchUserV3', u'searchOrgV2', u'integrationAppRead', u'questionsetImportAPI', u'getUsersPrograms', u'getUsersSolutions', u'getTargetedSolutions', u'getSolutionReportInformation', u'userLocationScope', u'mandatoryUserProfileFields', u'observationListEvidences', u'locationListing', u'getTargetedEntity', u'getProjectDetails', u'getProjectEntities', u'getProjectDetailReport', u'getProjectPrograms', u'updateProject', u'createProject', u'getProjectSolutionDetails', u'getProjectStatus', u'shareProject', u'getProjectTemplatesDetails', u'listObservationEntities', u'observationSubmissionsList', u'observationQuestionList', u'updateObservationSubmission', u'createObservationSubmission', u'searchLocationEntities', u'addObservationEntity', u'observationSolutionList', u'surveyQuestionList', u'solutionsVerifyLink', u'updateSurveySubmission', u'observationReports', u'surveyListEvidences', u'surveyReport', u'observationSubmissionCount', u'getFileUploadUrls', u'listPlatformPrograms', u'listPlatformSolutions', u'getUserSegmentsUCI', u'createUserSegmentsUCI', u'updateUserSegmentsUCI', u'deleteUserSegmentsUCI', u'userSegmentsQueryBuilderUCI', u'searchUserSegmentsUCI', u'createAdapterUCI', u'getAllAdapterUCI', u'getAdapterUCI', u'deleteAdapterUCI', u'updateAdapterUCI', u'getAllTransformerUCI', u'createTransformerUCI', u'getTransformerUCI', u'updateTransformerUCI', u'getFormsTransformerUCI', u'uploadFormsTransformerUCI', u'createConversationLogicUCI', u'getAllConversationLogicUCI', u'getConversationLogicUCI', u'deleteConversationLogicUCI', u'updateConversationLogicUCI', u'createBotUCI', u'getAllBotUCI', u'getBotUCI', u'startBotUCI', u'pauseBotUCI', u'searchBotUCI', u'deleteBotUCI', u'updateBotUCI', u'getAllUserForBotUCI', u'searchBotByStartingMessage', u'gqlUCI', u'sendNotification', u'readNotificationFeed', u'updateNotificationFeed', u'deleteNotificationFeed', u'netcoreWhatsappInbound', u'VerifyLink', u'GetTemplateOrQuestions', u'CreateProjectFromTemplate', u'updateUserV3', u'createContentV2', u'updateContentV2', u'readContentV2', u'uploadUrlContentV2', u'uploadContentV2', u'copyContentV2', u'importContentV2', u'flagContentV2', u'acceptFlagContentV2', u'discardContentV2', u'retireContentV2', u'reviewContentV2', u'rejectContentV2', u'linkDialcodeContentV2', u'reserveDialcodeContentV2', u'collectionReview', u'collectionReject', u'acceptCollectionFlag', u'reserveDialcodeCollection', u'listNotificationTemplate', u'createNotificationTemplate', u'deleteNotificationTemplate', u'updateNotificationTemplate', u'updateActionNotificationTemplate', u'getActionNotificationTemplate', u'createRCCertificate', u'searchRCCertificate', u'getRCCertificate', u'getRCPublicKey', u'getReportDatasets', u'projectCertificateList', u'createRCProjectCertificate', u'getProjectRCCertificate', u'fetchAllForm', u'releaseDialcodeContentV2', u'releaseDialcodeCollection', u'publishContentV2', u'unlistedPublishContent', u'unlistedPublishContentV2', u'publishCollection', u'unlistedPublishCollection', u'readDIALCodesBatchInfo']
Existing APIs : [u'PrivateContentUpdateAPIs', u'privateUserUpdate', u'PrivateContentReadAPIs', u'publishContent', u'getUserProfile', u'readCertTemplate', u'createPage', u'readContent', u'readContentState', u'readCourse', u'readFramework', u'readFrameworkTerm', u'readNote', u'readOrg', u'readUserProfile', u'registerMobileApp', u'registerMobileDevice', u'removeMember', u'retireContent', u'searchContent', u'searchCourse', u'searchNotes', u'searchOrg', u'searchUser', u'updateContentState', u'updateCourse', u'updateNote', u'updateOrg', u'updateOrgStatus', u'updatePage', u'updatePageSection', u'updateUser', u'uploadContent', u'userFeed', u'publishReport', u'retireReport', u'searchManagedUser', u'createGroup', u'updateGroup', u'listGroup', u'readGroup', u'programNominationUpdate', u'projectFeedSearch', u'questionCopy', u'questionBulkUpload', u'collectionHierarchyRead', u'createUser', u'echo', u'createUserVersion2', u'getUserProfileV2', u'listPageSections', u'getPageSettings', u'dialAssemble', u'assessmentItemSetRead', u'assessmentItemRead', u'searchProgram', u'questionsetCopy', u'createCertTemplate', u'programContenttypesList', u'createChannel', u'createFramework', u'createSession', u'orgAssignKeys', u'createAsset', u'programNominationList', u'updateAsset', u'programCollectionLink', u'publishFramework', u'qrCodeBatchProcessRetry', u'qrCodeBatchProcessStatus', u'readChannel', u'contentPrivateRead', u'readFrameworkCategory', u'retireLock', u'searchFrameworkCategory', u'searchFrameworkTerm', u'telemetryErrorLogging', u'updateCertTemplate', u'updateChannel', u'updateDialcode', u'updateFramework', u'updateFrameworkTerm', u'groupActivityAgg', u'listForm', u'getUserProfileV3', u'updateUserDeclarations', u'updateUserConsent', u'generateCertificate', u'getBatch', u'getUserType', u'indexSync', u'listChannel', u'listDialcodes', u'listFramework', u'listLock', u'listSystemSettings', u'listTerms', u'masterLocationCreate', u'masterLocationDelete', u'pluginSearch', u'privateUserRead', u'publishDialcode', u'searchChannel', u'questionBulkUploadStatus', u'programNominationAdd', u'programNominationRemove', u'compositePrivateSearch', u'assignRole', u'fetchURLMetaInformation', u'getAllIssuer', u'getAssertion', u'getAssertionList', u'blockUser', u'contentBadgeUnLink', u'copyContent', u'getAudienceCount', u'getBadgeClass', u'getCourseConsumptionDashboard', u'getCourseProgressDashboard', u'getCourseProgressDashboardV2', u'getUserConsumptionDasbhoard', u'getUserSkill', u'mergeUserAccounts', u'orgBulkUpload', u'PrivateContentCreateAPIs', u'readDialcode', u'createBadgeClass', u'createCourse', u'removeUserFromBatch', u'searchDialcodes', u'sendEmailNotification', u'sendNotification', u'unblockUser', u'updateCollaborators', u'createissuer', u'uploadMedia', u'userBulkUpload', u'readUserConsent', u'submitDataExhaustRequest', u'getDataExhaustRequest', u'listDataExhaustRequest', u'createTenantPreferences', u'updateTenantPreferences', u'readTenantPreferencess', u'generateCertificateV2', u'readAsset', u'createForm', u'createNote', u'deleteBadgeClass', u'deleteIssuer', u'deleteNote', u'flagContent', u'generateDialcodes', u'readPublisher', u'rejectContent', u'updatePublisher', u'updateUserSkill', u'updateForm', u'readForm', u'experimentread', u'generateOtp', u'addUserSkillEndorsement', u'deviceRegister', u'createUserVersion3', u'getData', u'getLocation', u'getSkills', u'KongConsumerApi', u'listOrdinals', u'listResourceBundles', u'onboardConsumer', u'profileVisibility', u'qrCodeDownload', u'rejectContentFlag', u'revokeAssertion', u'searchBadgeClass', u'searchData', u'submitContentForReview', u'telemetry', u'updateBatch', u'updateContent', u'updateCourseHierarchy', u'updateData', u'updateFrameworkCategory', u'updateLocation', u'uploadTOC', u'userExistenceApi', u'verifyOtp', u'addRegCertificateV2', u'downloadRegCertificateV2', u'druidGetUpdateListReport', u'druidDeactivateReport', u'druidCreateReport', u'readObjectCategory', u'uploadAsset', u'addEntity', u'updateEntity', u'readProgram', u'createProgram', u'updateProgram', u'addSystemSettings', u'batchBulkEnrolment', u'createAssertion', u'createData', u'createLocation', u'createUserLearnerV3', u'deleteData', u'deleteLocation', u'deviceProfile', u'experimentcreate', u'getAllData', u'searchEntity', u'listAllBadges', u'getUploadJobStatusLink', u'listOrgTypes', u'masterLocationSearch', u'checkUploadJobStatus', u'addUserToBatch', u'publishCourse', u'readCourseEnrollment', u'readPageSection', u'readRoleMapping', u'refreshLock', u'registerMobileDeviceTeacherAid', u'releaseDialcode', u'reserveDialcode', u'retireCourse', u'submitCourseForReview', u'updateDesktopApp', u'updateOrgType', u'updateUserloginTime', u'userMigrate', u'validateRegCertificate', u'verifyRegCertificate', u'updateObjectCategory', u'createObjectCategoryDefinition', u'readObjectCategoryDefinition', u'updateObjectCategoryDefinition', u'uploadUrlAsset', u'readEntity', u'deleteEntity', u'programCollectionCopy', u'programConfigurationSearch', u'programConfigurationList', u'downloadProgramList', u'programContentReport', u'programPublishContent', u'programPreferenceCreate', u'programPreferenceUpdate', u'mvcSearch', u'compositeSearch', u'courseEnrolment', u'createBatch', u'createFrameworkCategory', u'createLock', u'createOrgType', u'downloadTOC', u'getCourseHierarchy', u'getSystemSettings', u'ItemSetAPIs', u'LicenseAPI', u'listCourseEnrollments', u'listProgram', u'PrivateContentRejectAPI', u'PrivateContentReviewAPI', u'PrivateContentHierarchyAPI', u'PrivateContentPublishAPI', u'registerDesktopApp', u'uploadCrashLogs', u'UserBulkUploadv2', u'registerIgotApp', u'registerIgotDevice', u'listReports', u'getReport', u'updateReport', u'deleteReport', u'createReport', u'createReportSummary', u'getReportSummary', u'listReportSummary', u'searchRegCertificate', u'readRegCertificate', u'forgotPassword', u'questionCreate', u'questionRead', u'questionPrivateRead', u'questionUpdate', u'mvcContentCreate', u'questionSetRemoveQuestion', u'courseUnEnrolment', u'downloadCertificate', u'freshDesk', u'getMediaTypes', u'getParticipants', u'getTenantInfo', u'getUserByKey', u'masterLocationUpdate', u'programTenantList', u'masterLocationUpload', u'metricsSearchData', u'PrivateContentRetireAPI', u'programRegistrySync', u'bulkJobProcessRead', u'bulkJobProcessUpdate', u'contentImportAPI', u'retireAsset', u'publishProgram', u'unlistPublishProgram', u'bulkJobProcessCreate', u'PrivateContentUploadAPI', u'programPreferenceRead', u'questionSetReject', u'createPublisher', u'acceptContentFlag', u'addBadgeToUser', u'addMember', u'contentBadgeLink', u'createContent', u'linkDialcodeContent', u'downloadProgramNominationList', u'questionRetire', u'questionReview', u'questionPublish', u'questionReject', u'acceptTermsAndCondition', u'addRegCertificate', u'assemblePage', u'batchBulkUnEnrolment', u'createFrameworkTerm', u'createOrg', u'createPageSection', u'downloadRegCertificate', u'bulkJobProcessSearch', u'questionSetCreate', u'questionSetRead', u'questionSetPrivateRead', u'questionSetUpdate', u'questionSetRetire', u'questionSetReview', u'questionSetPublish', u'questionSetUpdateHierarchy', u'questionSetReadHierarchy', u'questionSetAddQuestion', u'collectionPrivateRead', u'addUserSkill', u'copyFramework', u'courseBatchRemoveCertificateTemplate', u'registerMobileAppTeacherAid', u'contributorSearch', u'courseBatchAddCertificateTemplate', u'courseIssueCertificate', u'getContentUploadUrl', u'getIssuerDetails', u'privateUserAssignRole', u'registerMobileDeviceOpenRAP', u'copyAsset', u'createContentV2', u'collectionRetire', u'collectionHierarchyAdd', u'collectionHierarchyRemove', u'collectionHierarchyUpdate', u'collectionCopy', u'collectionImport', u'collectionExport', u'collectionReview', u'collectionReject', u'acceptCollectionFlag', u'registerDesktopDevice', u'registerMobileAppOpenRAP', u'questionList', u'updateContentV2', u'readContentV2', u'uploadUrlContentV2', u'uploadContentV2', u'channelSpecificTelemetryExhaust', u'createObjectCategory', u'copyContentV2', u'importContentV2', u'flagContentV2', u'acceptFlagContentV2', u'discardContentV2', u'retireContentV2', u'reviewContentV2', u'rejectContentV2', u'collectionCreate', u'collectionRead', u'collectionUpdate', u'collectionFlag', u'collectionDiscard']
Adding API assignRoleV2
{"strip_uri":"strip_uri is an unknown field","uris":"uris is an unknown field","request_path":"At least a 'request_host' or a 'request_path' must be specified","request_host":"At least a 'request_host' or a 'request_path' must be specified"}

STDERR:

WARNING:retry.api:HTTP Error 400: Bad Request, retrying in 2 seconds...
WARNING:retry.api:HTTP Error 400: Bad Request, retrying in 4 seconds...
WARNING:retry.api:HTTP Error 400: Bad Request, retrying in 8 seconds...
WARNING:retry.api:HTTP Error 400: Bad Request, retrying in 16 seconds...
Traceback (most recent call last):
  File "/tmp/kong-api-scripts/kong_apis.py", line 82, in <module>
    save_apis(args.kong_admin_api_url, input_apis)
  File "/tmp/kong-api-scripts/kong_apis.py", line 24, in save_apis
    json_request("POST", apis_url, _sanitized_api_data(input_api))
  File "/tmp/kong-api-scripts/common.py", line 36, in json_request
    response = retrying_urlopen(request)
  File "</usr/local/lib/python2.7/dist-packages/decorator.pyc:decorator-gen-2>", line 2, in retrying_urlopen
  File "/usr/local/lib/python2.7/dist-packages/retry/api.py", line 74, in retry_decorator
    logger)
  File "/usr/local/lib/python2.7/dist-packages/retry/api.py", line 33, in __retry_internal
    return f()
  File "/tmp/kong-api-scripts/common.py", line 41, in retrying_urlopen
    return urllib2.urlopen(*args, **kwargs)
  File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 400: Bad Request

MSG:

non-zero return code`

**5)KnowledgeMW**
`fatal: [localhost]: FAILED! => {}

MSG:

The task includes an option with an undefined variable. The error was: 'adminutil_refresh_token_public_key_kid' is undefined

The error appears to be in '/var/lib/jenkins/workspace/Deploy/DockPreproduction/Kubernetes/KnowledgeMW/kubernetes/ansible/roles/sunbird-deploy/tasks/main.yml': line 42, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

- name: Create the token public key file
  ^ here`

**6)KafkaSetup**
`failed: [11.6.2.8] (item={'retention_time': 604800000, 'name': 'assessment.republish.request', 'num_of_partitions': 1, 'replication_factor': 1}) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": [
        "/opt/kafka/bin/kafka-topics.sh",
        "--zookeeper",
        "localhost:2181",
        "--alter",
        "--topic",
        "preproddock.assessment.republish.request",
        "--config",
        "retention.ms=604800000"
    ],
    "delta": "0:00:01.288795",
    "end": "2023-01-20 18:23:30.161817",
    "item": {
        "name": "assessment.republish.request",
        "num_of_partitions": 1,
        "replication_factor": 1,
        "retention_time": 604800000
    },
    "rc": 1,
    "start": "2023-01-20 18:23:28.873022"
}

STDOUT:

Error while executing topic command : Topic preproddock.assessment.republish.request does not exist on ZK path localhost:2181
`

**Comment by gandham-santhosh** [comment]
@sandeep-r1 can you check if all the variables that the job is looking for are initialised in the common.yaml and are being passed properly?
@santhosh-tg Please check this

**Reply by Krishnaj20** [reply]
@sandeep-r1 @santhosh-tg - Is your query resolved? If so, pls mark it as answered.

**Comment by Krishnaj20** [comment, marked as answer]
@sandeep-r1 - this query is resolved in https://github.com/orgs/Sunbird-cokreat/discussions/42. Hence marking this as answered.
